pyanaconda/ui/lib/subscription.py

```diff
@@ -59,8 +60,8 @@ def _tear_down_existing_source(payload):
     :param payload: Anaconda payload instance
     """
     source_proxy = payload.get_source_proxy()
-
-    if source_proxy.Type == SOURCE_TYPE_HDD and source_proxy.Partition:
+    should_unamrk_protected = source_proxy.Type == SOURCE_TYPE_HDD and source_proxy.Partition
+    if payload.type == PAYLOAD_TYPE_DNF and should_unamrk_protected:
```

Comment:
I don't get it. The payload type is already checked before you call `switch_source` that calls ` _tear_down_existing_source`, so the type should be correct at this point.

Reply:
Yeah, you are right - I don't think we need to be that careful if it's checked before `switch_source()`. I'll drop that. :)